reconcile/utils/gitlab_api.py

```diff
@@ -250,6 +251,46 @@ def get_group_if_exists(self, group_name: str) -> Group | None:
         except gitlab.exceptions.GitlabGetError:
             return None
 
+    def share_project_with_group(
+        self, repo_url: str, group_id: int, dry_run: bool, access: str = "maintainer"
+    ) -> None:
+        gitlab_request.labels(integration=INTEGRATION_NAME).inc()
+        project = self.get_project(repo_url)
+        if project is None:
+            return None
+        access_level = self.get_access_level(access)
+        gitlab_request.labels(integration=INTEGRATION_NAME).inc()
+        # check if we have 'access_level' access so we can  add the group with same role.
+        members = self.get_items(
+            project.members.all, query_parameters={"user_ids": self.user.id}
+        )
+        if self.user.id not in [
+            member.id for member in members if member.access_level >= access_level
+        ]:
+            logging.error(
+                "'{}' is not shared with {} as 'Maintainer'".format(
+                    repo_url, self.user.username
+                )
+            )
```

Comment:
shouldn't hard code `Maintainer` as it's can be other `access`, and no need to call `format` in `logging.error`

```python
            logging.error(
                "%s is not shared with %s as %s",
                repo_url,
                self.user.username,
                access,
            )
```